From 2cb57a911f67c8c6574789b546194831554830a3 Mon Sep 17 00:00:00 2001
From: Alejandro Abdelnur <tucu@apache.org>
Date: Fri, 15 Mar 2013 17:48:32 +0000
Subject: [PATCH 1298/1357] MR1: MAPREDUCE-5072. TestDelegationTokenRenewal.testDTRenewal fails in MR1 on jdk7. (sandyr via tucu)

git-svn-id: https://svn.apache.org/repos/asf/hadoop/common/branches/branch-1@1457034 13f79535-47bb-0310-9956-ffa450edef68
(cherry picked from commit d473428164658260bdc691f047c7411d98a946c2)

Reason: Fix failing jdk7 tests
Ref: CDH-11028
Author: Sandy Ryza
---
 .../security/token/TestDelegationTokenRenewal.java |   13 ++++++++++---
 1 files changed, 10 insertions(+), 3 deletions(-)

diff --git a/hadoop-mapreduce1-project/src/test/org/apache/hadoop/mapreduce/security/token/TestDelegationTokenRenewal.java b/hadoop-mapreduce1-project/src/test/org/apache/hadoop/mapreduce/security/token/TestDelegationTokenRenewal.java
index 29ec8ee..5c73ee5 100644
--- a/hadoop-mapreduce1-project/src/test/org/apache/hadoop/mapreduce/security/token/TestDelegationTokenRenewal.java
+++ b/hadoop-mapreduce1-project/src/test/org/apache/hadoop/mapreduce/security/token/TestDelegationTokenRenewal.java
@@ -47,7 +47,7 @@ import org.apache.hadoop.security.token.Token;
 import org.apache.hadoop.security.token.SecretManager.InvalidToken;
 import org.apache.hadoop.security.token.TokenRenewer;
 import org.apache.hadoop.util.StringUtils;
-import org.junit.BeforeClass;
+import org.junit.Before;
 import org.junit.Test;
 
 /**
@@ -65,6 +65,12 @@ public class TestDelegationTokenRenewal {
     private static Token<?> lastRenewed = null;
     private static Token<?> tokenToRenewIn2Sec = null;
 
+    private static void reset() {
+      counter = 0;
+      lastRenewed = null;
+      tokenToRenewIn2Sec = null;
+    }
+    
     @Override
     public boolean handleKind(Text kind) {
       return KIND.equals(kind);
@@ -107,8 +113,9 @@ public class TestDelegationTokenRenewal {
   private static Configuration conf;
   private static String trackerService = "localhost:0";
  
-  @BeforeClass
-  public static void setUp() throws Exception {
+  @Before
+  public void setUp() throws Exception {
+    Renewer.reset();
     conf = new Configuration();
     conf.set("mapred.job.tracker", trackerService);
     
-- 
1.7.0.4

